Forward3

eirnym opened the pull request:
Rebased version of `forward` branch. Could be placed on top of master branch of [pyeve](https://github.com/pyeve/cerberus) and [this](https://github.com/funkyfuture/cerberus) repositories

funkyfuture committed:
Bumps version to 2.0.dev0

funkyfuture committed:
Renames the 'schema' rule for sequences to 'itemsrules'

funkyfuture committed:
Simplifies some names in the errors module

funkyfuture committed:
Makes use of set literals and dictionary comprehensions

funkyfuture committed:
Adds type annotations to the validator module

funkyfuture committed:
Adds type annotations to the errors module

funkyfuture committed:
Adds type annotations to the base module

funkyfuture committed:
Removes deprecated rule names

funkyfuture committed:
Removes type checking per methods

funkyfuture committed:
Removes support for Python 2

funkyfuture committed:
Employs Python 3's class machinery

funkyfuture committed:
Facilitates machinery and requirements for type annotations of the code

In order to use literal type annotations, support for CPython 3.4 was
removed which will be end-of-life before CPython 2.7.

funkyfuture committed:
Adds type annotations to the utils module

funkyfuture committed:
Overhauls the mechanics of schema validation

Yippie, that makes the code more lightweight and much easier to
understand.

funkyfuture committed:
Adds type annotations to the schema module

eirnym commented:
I'm wondering why did you removed Python 3.7 in your commits.

funkyfuture commented:
okay, i updated the master branch in this repo to align with the official one. yet, github doesn't update this pr accordingly, hence the commits 28e1ee7..688a67a are still appearing here. they must not.

i also set up automated tests here.

> I'm wondering why did you removed Python 3.7 in your commits.

could you please provide a hyperlink to what you mean?

eirnym commented:
I rebased to the `https://github.com/pyeve/cerberus/tree/master`, not to the `https://github.com/funkyfuture/cerberus/tree/master`. Probably I've seen [this](https://github.com/funkyfuture/cerberus/commit/0f7a2a5037cab1f1441b4d330220bd4804cbc373) commit as [this](https://github.com/funkyfuture/cerberus/pull/3/commits/264d0e595fdea80b8b85dde371dfdbaeb0dac84c)

eirnym commented:
Should I create a pull request to master repo?

funkyfuture commented:
> Should I create a pull request to master repo?

doesn't matter for now since i updated here. but just a new pr that *compares* (that is something else than rebasing) to the current master and triggers the tests.

eirnym commented:
A few tests are not passing, as the internals were changed, and information about what's is wrong in a schema is not returning out with an exception.

https://github.com/pyeve/cerberus/pull/479

eirnym commented:
Fill free to edit my branch as you have an access

funkyfuture commented:
as i said, ain't that easy. e.g. blindly reapplying the changes to `.travis.yml`.

eirnym commented:
This is why I did it with merge first.

eirnym committed:
Add back missing references to pkg_resources

eirnym committed:
Fix regex syntax

eirnym committed:
Fix pytest.raises() deprecation warning

Please comment on https://github.com/pytest-dev/pytest/issues/3974 if you have concerns about removal of this parameter.

eirnym committed:
Add Py37 back in tox.ini and .travis  and .travis.yml

eirnym committed:
Fix rebasing

eirnym committed:
One more schema

eirnym commented:
Fixed tests, warnings in tests, .travis.yml and tox.ini

Locally run tests on python3.7

eirnym commented:
I don't know what's wrong with `doctest` tox environment. Locally it works like a charm. Should I "return" `matrix` to `travis.yml`?

eirnym commented:
@funkyfuture Could you please check if any of these changes are needed. Please, close this PR is it's too old to be included
